### ISSUE TITLE
Unable to connect API to a proxy in nightly APInf

### ISSUE THREAD

**Issue by Nazarah** [apinf/platform#3228]
**Steps to Reproduce**: 
1. go to page: https://nightly.apinf.io and login
2. Click Add API and start adding an API manually (used the following endpoint URL: https://nightly.apinf.io)
3. After adding the API, go to proxy tab in API profile.
4. Select Nightly Umbrella as proxy and add '/v/' as proxy base path and '/ ' as API base path.
5. Click  Save and publish
                    
**Outcome**:
Proxy set up doesn't get saved.
s-alert shows error: "can't read property length of null value".
If tried to change the selected proxy, the proxy name didn't change in the respected field.
If tried to save and publish by changing the proxy, it failed.
S alert showed error: failed [500] {"status"."500","error","Internal Server Error"}
http://recordit.co/lbmjMT5RwW
         
**Expected Behavior**:

added values for proxy base path and api base path should be saved accordingly.
User should be able to see Get API Key button upon publishinbg.
                   
**Environment**:

Windows 10, Chrome browser (latest), APInf nightly (https://nightly.apinf.io)

**Comment by Nazarah** [comment]
Just to add a note, apinf staging with 0.52 release is really actin wierd.
I discovered the bug as part of testing 0.52 in staging and came to nightly to verify the existence.
Later when I wanted to record the bug in staging, I was able to save and publish the API by setting proxy set path.
So the bug was non-producible in APInf staging in later attempts.
The similar situation happened for issue  #3227
